### PR TITLE
Replace chrono with time

### LINKED ISSUE #171: Remove dependency on chrono?
```
Crate:         chrono
Version:       0.4.19
Title:         Potential segfault in `localtime_r` invocations
Date:          2020-11-10
ID:            RUSTSEC-2020-0159
URL:           https://rustsec.org/advisories/RUSTSEC-2020-0159
Solution:      No safe upgrade is available!
Dependency tree: 
chrono 0.4.19
├── yup-oauth2 6.3.1
│   └── gcp-bigquery-client 0.11.0
│       └── cjms 1.0.0
└── gcp-bigquery-client 0.11.0
```

chrono does not appear to be actively maintained.

Could the `time` crate meet the needs of this repo?

### PR DESCRIPTION
**Pull request by birdsarah**
Fixes #171

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,24 +31,24 @@ default = ["hyper-rustls", "service_account"]
 service_account = ["rustls"]
 
 [dependencies]
+anyhow = "1.0.38"
+async-trait = "^0.1"
```
Comment (birdsarah) [inline Cargo.toml:35]: sorry i couldn't help myself with the alphabetic ordering. let me know if you want me to do this as a separate PR.

File: src/authenticator_delegate.rs [hunk 2]
```diff
@@ -94,10 +94,11 @@ async fn present_user_code(device_auth_resp: &DeviceAuthResponse) {
...
     );
     println!("Do not close this application until you either denied or granted access.");
-    println!(
-        "You have time until {}.",
-        device_auth_resp.expires_at.with_timezone(&Local)
-    );
+    let printable_time = match UtcOffset::current_local_offset() {
+        Ok(offset) => device_auth_resp.expires_at.to_offset(offset),
+        Err(_) => device_auth_resp.expires_at,  // Fallback to printing in UTC
+    };
```
Comment (birdsarah) [inline src/authenticator_delegate.rs:100]: I don't know if this works as you want it to and I don't think you have a test for it. You may want to verify.

File: Cargo.toml (hunk 1 above)
Comment (dermesser) [reply Cargo.toml:35]: no, on the contrary - thank you for cleaning up!

File: src/authenticator_delegate.rs (hunk 2 above)
Comment (dermesser) [reply src/authenticator_delegate.rs:100]: It doesn't seem to be able to determine my timezone and falls back to UTC; the previous version handled this correctly. It isn't a blocker for me personally, as I believe this feature to only be used rarely (and the displayed fact is still correct). Nevertheless, if you know a typical reason for `time` to not be able to find the timezone (I also tested just displaying the local time, which also didn't work) off the top of your head, this would be the cherry on top.

**Review by dermesser — APPROVED** [review]

File: src/authenticator_delegate.rs (hunk 2 above)
Comment (birdsarah) [reply src/authenticator_delegate.rs:100]: I'm afraid I don't know why it wouldn't be working for you. Sorry :( The implementation does try to cover all platforms - https://github.com/time-rs/time/tree/main/src/sys/local_offset_at

Comment (dermesser) [reply src/authenticator_delegate.rs:100]: That's perfectly fine, thank you for the whole effort in any case! :)
